### PR TITLE
wence/fix/bc initialisation

### PR DESCRIPTION
**Pull request by wence-**
Simplifies the logic for getting bc values into the target space. Also catches a load of bugs around non-homogenous bc values for vector-valued spaces.

### CODE REVIEW

File: tests/slate/test_hdg_poisson.py [hunk 1]
```diff
@@ -93,7 +93,6 @@ def run_LDG_H_problem(r, degree, quads=False):
     return scalar_error, flux_error
 
 
-@pytest.mark.parallel
```
Comment (dham) [inline tests/slate/test_hdg_poisson.py:96]: Why is this removed?

Comment (wence-) [reply tests/slate/test_hdg_poisson.py:96]: I don't recall, I'll drop it.

Comment (wence-) [reply tests/slate/test_hdg_poisson.py:96]: Possibly leftover from debugging the test.
